packages/react-devtools-extensions/src/background.js

```diff
@@ -70,14 +72,28 @@ function installProxy(tabId: number) {
   }
 }
 
-function doublePipe(one, two) {
+function doublePipe(one, two, tabId) {
   one.onMessage.addListener(lOne);
   function lOne(message) {
-    two.postMessage(message);
+    try {
+      two.postMessage(message);
+    } catch (e) {
+      if (__DEV__) {
+        console.log(`Broken pipe ${tabId}: `, e);
+      }
+      shutdown();
```

Comment:
Would the postMessage fail because of a JS exception?  I'm not sure if it will happen. I think if we shutdown here not during a page load, the devtools would stop working?

Reply:
It's not a JS exception. It's a service worker exception when `postMessage` is fired on a tab that's offloaded. The error message looks like this:
<img width="627" alt="image" src="https://user-images.githubusercontent.com/1001890/216444811-8c54ea90-356e-402d-a3ca-5e8e40fec962.png">

It's weird that the listener was not removed since we already have `one.onDisconnect.addListener(shutdown)`. Maybe Chrome doesn't treat reload as disconnect anymore?